app/common/utils.js

```diff
@@ -0,0 +1,7 @@
+export function shiftItemRandomly(array) {
+    return array.splice(getRandomInt(0, array.length), 1)[0];
```

Comment:
Pop implicitly refers to the last element of a stack. So does shift (in ES6), but at least *explicitly* means "move from one place to another". Perhaps the simplest solution would be "shiftItemRandomly_**from**_array, to make clear the intention of extracting an element from an array, thus removing it.